### PR TITLE
Regressive tests

### PR DESCRIPTION
**Pull request by dalderliesten**
Refactored multiple classes, refactored many test classes, fixed many checkstyle issues, and added many tests to increase the overall coverage.

### CODE REVIEW

File: src/test/java/nl/tudelft/pixelperfect/SpaceshipTest.java [hunk 1]
```diff
@@ -27,39 +28,53 @@ public void before() {
   }
 
   /**
-   * Test the get health functionality by verifying full health.
+   * Test the getLog functionality by checking the existance of a log.
+   */
+  @Test
+  public void testGetLog() {
+
```
Comment (dalderliesten) [inline src/test/java/nl/tudelft/pixelperfect/SpaceshipTest.java]: This test is empty until further notice, due to the current abscence of log entries.

**Comment by wouterz** [comment]
Looks good, but a bit excessive. Don't go overboard with refactoring everything the way you would have done it and renaming things ever so slightly. Different people will have different styles or naming habits. As long as its clear, leave it (imo)

**Comment by dalderliesten** [comment]
Wouter, many of the changed variables/names were changed due to checkstyle errors.  For example, the "s" variable gave me two checkstyle issues, as it does not follow proper convention.

**Comment by wouterz** [comment]
What kind of checkstyle error? never had any variable/naming errors unless using faulty capitalization (and still dont now when i run my checkstyle using the projects checkstyle rules)

**Comment by jessetilro** [comment]
There are some conflicts with the release branch. You should pull release into your branch and solve possible merge conflicts.

File: src/main/java/nl/tudelft/pixelperfect/client/ConnectListener.java [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
-   * @param server The server it refers to.
-   * @param client The client that's trying to disconnect.
-   */
-  public void connectionRemoved(Server server, HostedConnection client) {
-    System.out.println("Client " + client + " has disconnected from the game");
-    
-  }
-
-}
+package nl.tudelft.pixelperfect.client;
```
Comment (jessetilro) [inline src/main/java/nl/tudelft/pixelperfect/client/ConnectListener.java]: Why exactly were all lines replaced in this file?

File: src/main/java/nl/tudelft/pixelperfect/route/RouteNode.java [hunk 3]
```diff
@@ -16,9 +16,9 @@
    * A constructor for the RouteNode.
    * 
    * @param summary
-   *          , a summary about the RouteNode.
+   *          a summary about the RouteNode.
```
Comment (jessetilro) [inline src/main/java/nl/tudelft/pixelperfect/route/RouteNode.java]: While you're at it, sentences in javadoc should be capitalized as well.

File: src/main/java/nl/tudelft/pixelperfect/client/ConnectListener.java (hunk 2 above)
Comment (dalderliesten) [inline src/main/java/nl/tudelft/pixelperfect/client/ConnectListener.java]: I find that very strange, I only edited the comments on javadoc, and the editoir automatically added by Authoring name due to that.  Are you able to notice any differences?

**Comment by dalderliesten** [comment]
Working on the merge conflicts, they arsose as of the accepted PR from a few moments ago.

**Comment by dalderliesten** [comment]
#57 is related to this PR.

**Comment by jessetilro** [comment]
I will now fix the merge conflicts.

**Pull request merged**